test/metabase/api/session_test.clj

```diff
@@ -34,18 +34,17 @@
 
 (use-fixtures :once (fixtures/initialize :db :web-server :test-users))
 
-(use-fixtures :each (fn [thunk]
-                      ;; reset login throtllers
-                      (doseq [throttler (vals @#'api.session/login-throttlers)]
-                        (reset! (:attempts throttler) nil))
-                      (thunk)))
+(defn- reset-throttlers! []
+  (doseq [throttler (vals @#'api.session/login-throttlers)]
+    (reset! (:attempts throttler) nil)))
 
 (def ^:private SessionResponse
   {:id (s/pred mt/is-uuid-string? "session")})
 
 (def ^:private session-cookie @#'mw.session/metabase-session-cookie)
 
 (deftest login-test
+  (reset-throttlers!)
```

Comment:
It's a limitation of the linter I wrote that it can't detect when when you put thread-unsafe stuff in the fixtures and then be able to error when you use `^:parallel` form, so for now I just disallowed thread-unsafe stuff in the fixtures. Hopefully that will encourage people to figure out how to write thread-safe tests. I think we could make those throttlers dynamic and parallelize almost this whole namespace in the future.

Reply:
OK, I was thinking about marking all tests in the namespace as sequential if we detect a thread-unsafe call in a for-each fixture.
But I do see the point about encouraging people to strive for parallel tests.

Reply:
Yeah unfortunately I don't think you can do that with custom Kondo linters at this point, since you can only write static linters for individual forms and don't have access to whatever is happening in the rest of the namespace, so there would be no way to know if that namespace had test fixtures defined or not.